## ETL sobre el archivo review.json de la plataforma Yelp

***Importamos las librerias***

In [1]:
import os
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from wordcloud import WordCloud
import re

***Para no sobrecargar la memoria e importar los datos correctamente, lo hacemos en chunks.***

In [2]:
# Ruta del archivo
file_path = r'C:\Users\Usuario\Desktop\HENRY\PROYECTO FINAL\Proyecto Final\datasets\Yelp\review.json'

# Leer el archivo JSON en chunks
chunksize = 150000  # Puedes ajustar este valor según sea necesario

# Crear un iterador para leer el archivo en chunks
json_reader = pd.read_json(file_path, lines=True, chunksize=chunksize)

# Inicializar un DataFrame vacío
df_reviews_y = pd.DataFrame()

# Iterar sobre los chunks y procesar cada uno
for chunk in json_reader:
    # Procesar el chunk
    df_reviews_y = pd.concat([df_reviews_y, chunk])

***Visualizamos la cantidad de registros***

In [3]:
len(df_reviews_y)

6990280

***Filtramos el Dataset de las reviews, vinculandolo con el dataset de locales y dejando asi, solamente las reviews que sean de estado de California***

In [5]:
# Ruta del archivo Parquet
parquet_path = r'C:\Users\Usuario\Desktop\HENRY\PROYECTO FINAL\repo\epicurean_project\DATA\data parcial\locales_gastronomicos_yelp.parquet'

# Cargar la columna 'business_id' desde el archivo Parquet
busines_id_locales = pd.read_parquet(parquet_path, columns=['business_id'])

# Filtrar los registros que coincidan en la columna 'business_id'
df_reviews_y = df_reviews_y[df_reviews_y['business_id'].isin(busines_id_locales['business_id'])]

***Visualizamos nuevamente la cantidad de registros obtenida***

In [6]:
len(df_reviews_y)

214

Como podemos observar, sólo contamos con 214 registros, lo que torna estos datos en irrelevantes de cara a nuestro análisis.